### ISSUE TITLE
NullPointerException?

### ISSUE THREAD

**Issue by PokezardVGC** [nus-cs2030/2122-s1#224]
Realising that we call do item == null check, created a new Optional.of(item) and checking if it equals to Optional.empty()
Output as shown, which seems quite baffling
If I check for item instanceofLogger in the else if statement, that statement will be NullPointerException. 
Anyone has an explanation to this?
![Uploading image.png…]()

**Comment by RussellDash332** [comment]
Since the optional can take in `null`, you should use `Optional.ofNullable` instead of `Optional.of`

**Comment by PokezardVGC** [comment]
Hmm, but if null can be taken in as a value to Optional.<U>ofNullable, will it make a difference if we use that as opposed to Optional<U>.of only?

**Comment by RussellDash332** [comment]
If you see the Javadoc, the only difference between `Optional.of` and `Optional.ofNullable` is that given `null` as input, the former throws NPE and the latter returns an empty `Optional`.

**Comment by PokezardVGC** [comment]
Thanks so much sir, time to actually read the docs in detail

**Comment by RussellDash332** [comment]
> Thanks so much sir, time to actually read the docs in detail

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html

**Comment by PokezardVGC** [comment]
If anyone else is encountering the game problem, its with the spelling of your output. Spelt "argument" as "arguement". Think code crunch only allows the exception of "argument cannot be null", any misspellings will not be flagged out as wrong answer but null value detected

**Comment by lulucopter** [comment]
Yes, I suspect code crunch only allows any intended mention of the word null. In this case, by spelling argument correctly the exception not be flagged as null by code crunch
